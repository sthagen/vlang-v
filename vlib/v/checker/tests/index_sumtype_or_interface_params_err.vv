type Foo = int | map[int]int

interface Bar {}

fn sum_type_or_interface_as_parameters(mut foo Foo, mut bar Bar) {
	_ = foo[0]
	_ = bar[0]
}
