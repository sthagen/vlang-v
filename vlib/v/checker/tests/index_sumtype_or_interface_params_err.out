vlib/v/checker/tests/index_sumtype_or_interface_params_err.vv:6:9: error: type `Foo` does not support indexing
    4 | 
    5 | fn sum_type_or_interface_as_parameters(mut foo Foo, mut bar Bar) {
    6 |     _ = foo[0]
      |            ~~~
    7 |     _ = bar[0]
    8 | }
vlib/v/checker/tests/index_sumtype_or_interface_params_err.vv:7:9: error: type `Bar` does not support indexing
    5 | fn sum_type_or_interface_as_parameters(mut foo Foo, mut bar Bar) {
    6 |     _ = foo[0]
    7 |     _ = bar[0]
      |            ~~~
    8 | }
